swarm/api/manifest.go

```diff
@@ -62,6 +62,11 @@ func readManifest(manifestReader storage.LazySectionReader, hash storage.Key, dp
 
 	// TODO check size for oversized manifests
 	size, err := manifestReader.Size(quitC)
+	if err != nil { // size == 0
+		// can't determine size means we don't have the root chunk
+		glog.V(logger.Detail).Infof("Manifest %v not found.", hash.Log())
+		return
```

Comment:
yes to descriptive error.
no need to log the error indeed, but we do it other places, maybe another PR can change these.
Also for the particular error returned, see #3447